sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/tests/Microsoft.Azure.WebJobs.Extensions.EventGrid.Tests.csproj

```diff
@@ -16,6 +16,7 @@
 
   <ItemGroup>
     <ProjectReference Include="$(AzureCoreTestFramework)" />
+    <ProjectReference Include="..\..\..\extensions\Microsoft.Extensions.Azure\src\Microsoft.Extensions.Azure.csproj" />
```

Comment:
Why do we need to change Event Grid?

Reply:
we include whole folder with shared classes in the same csproj:
https://github.com/Azure/azure-sdk-for-net/pull/35344/files#diff-5e23e83cf9e4a5e7ff08b8f72a47f198457ed95c8b115e8886c1ebe701019f3cR24

there's the new `TestComponentFactory` class which need reference to `Microsoft.Extensions.Azure` for `AzureComponentFactory`